source/Handlebars/Compiler/Lexer/Converter/BlockAccumulators/BlockAccumulatorContext.cs

```diff
@@ -77,6 +81,54 @@ private static bool IsPartialBlock (Expression item)
             }
         }
 
+        private static bool IsLooseClosingElement(Expression item, Expression parentItem, out string looseBlockName)
+        {
+            looseBlockName = null;
+
+            var itemBlockName = GetBlockName(item);
+
+            if (itemBlockName == null) return false;
+
+            var parentBlockName = GetBlockName(parentItem);
+
+            if (!itemBlockName.StartsWith("/")) return false;
+
+            if (parentBlockName == null || IsClosingBlockNotMatchParentBlock(itemBlockName, parentBlockName))
+            {
+                looseBlockName = itemBlockName;
+
+                return true;
+            }
+
+            return false;
+        }
+
+        private static bool IsClosingBlockNotMatchParentBlock(string itemBlockName, string parentBlockName)
+        {
+            if (itemBlockName == null) throw new ArgumentNullException(nameof(itemBlockName));
+            if (parentBlockName == null) throw new ArgumentNullException(nameof(parentBlockName));
+
+            if (!parentBlockName.StartsWith("#") || parentBlockName.StartsWith("#>") || parentBlockName.StartsWith("#*")) return false;
+
+            return parentBlockName.Substring(1) != itemBlockName.Substring(1);
```

Comment:
Why can't you just do `parentBlockName[0] != itemBlockName[0]`? `Substring` will produce new string allocation.

Reply:
I'm not sure that I've understood you correctly because `parentBlockName[0] != itemBlockName[0]` will just compare "#" with "/" and it will cause a false positive alarm for any correct statement like `{{#if}}test{{/if}}`. Here the code compares if an open element matches with a closing element (e.g. `#if` with `/unless`).

Reply:
Yes, I'm aware of string allocations and the best I can do (as far as I know) is to use `MemoryExtensions.AsSpan` method but it's available only from .NET Standard 2.1. So should I use preprocessor directive `#if netstandard2_1 ...` for this case or do you know a better solution?

Reply:
> I'm not sure that I've understood you correctly because parentBlockName[0] != itemBlockName[0] will just compare "#" with "/"

My bad, I've interpreted the code in the wrong way. 

> Yes, I'm aware of string allocations and the best I can do (as far as I know) is to use `MemoryExtensions.AsSpan` method but it's available only from .NET Standard 2.1. So should I use preprocessor directive `#if netstandard2_1 ...` for this case or do you know a better solution?

You may use `Substring` struct:
```suggestion
            return new Substring(parentBlockName, 1) != new Substring(itemBlockName, 1);
```
Based on benchmarking it behaves equivalently from execution time but better in terms of allocations:

|                 Method |  N |      Mean |     Error |    StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|----------------------- |--- |----------:|----------:|----------:|-------:|------:|------:|----------:|
|            EqualString |  2 |  4.741 ns | 0.0186 ns | 0.0156 ns |      - |     - |     - |         - |
|            EqualString |  5 | 29.061 ns | 0.2653 ns | 0.2482 ns | 0.0204 |     - |     - |      64 B |
|            EqualString | 10 | 27.998 ns | 0.1122 ns | 0.0937 ns | 0.0255 |     - |     - |      80 B |
|                        |    |           |           |           |        |       |       |           |
|         EqualSubstring |  2 | 22.192 ns | 0.0824 ns | 0.0730 ns |      - |     - |     - |         - |
|         EqualSubstring |  5 | 20.849 ns | 0.0441 ns | 0.0413 ns |      - |     - |     - |         - |
|         EqualSubstring | 10 | 20.914 ns | 0.0739 ns | 0.0691 ns |      - |     - |     - |         - |
|                        |    |           |           |           |        |       |       |           |
|    NotEqualStartString |  2 | 26.196 ns | 0.1376 ns | 0.1220 ns | 0.0204 |     - |     - |      64 B |
|    NotEqualStartString |  5 | 26.828 ns | 0.4037 ns | 0.3371 ns | 0.0204 |     - |     - |      64 B |
|    NotEqualStartString | 10 | 29.938 ns | 0.2559 ns | 0.2393 ns | 0.0306 |     - |     - |      96 B |
|                        |    |           |           |           |        |       |       |           |
| NotEqualStartSubstring |  2 | 23.794 ns | 0.0685 ns | 0.0641 ns |      - |     - |     - |         - |
| NotEqualStartSubstring |  5 | 25.827 ns | 0.1304 ns | 0.1089 ns |      - |     - |     - |         - |
| NotEqualStartSubstring | 10 | 25.739 ns | 0.0882 ns | 0.0737 ns |      - |     - |     - |         - |
|                        |    |           |           |           |        |       |       |           |
|      NotEqualEndString |  2 | 28.473 ns | 0.2956 ns | 0.2765 ns | 0.0204 |     - |     - |      64 B |
|      NotEqualEndString |  5 | 28.210 ns | 1.7463 ns | 1.5481 ns | 0.0204 |     - |     - |      64 B |
|      NotEqualEndString | 10 | 30.028 ns | 0.3705 ns | 0.3466 ns | 0.0306 |     - |     - |      96 B |
|                        |    |           |           |           |        |       |       |           |
|   NotEqualEndSubstring |  2 | 26.405 ns | 0.0686 ns | 0.0608 ns |      - |     - |     - |         - |
|   NotEqualEndSubstring |  5 | 25.046 ns | 0.0434 ns | 0.0406 ns |      - |     - |     - |         - |
|   NotEqualEndSubstring | 10 | 29.412 ns | 0.1160 ns | 0.1085 ns |      - |     - |     - |         - |

Reply:
Thank you, I didn't know about this useful utility function. Updated!